java/common/exception/ErrorMessage.java

```diff
@@ -87,14 +87,16 @@ public class ErrorMessage extends com.vaticle.typedb.common.exception.ErrorMessa
             new ErrorMessage(31, "Rule '%s' 'then' '%s' tries to assign type '%s' to variable '%s', but this variable already had a type assigned by the rule 'when'. Try omitting this type assignment.");
     public static final ErrorMessage INVALID_RULE_THEN_VARIABLES =
             new ErrorMessage(32, "Rule '%s' 'then' variables must be present in rule 'when'.");
+    public static final ErrorMessage INVALID_RULE_THEN_ROLES =
+            new ErrorMessage(33, "Rule '%s' 'then' '%s': all roles must be specified explicitly or using a variable.");
```

Comment:
Looks like the style of these error messages is more "english"-y, using full sentences. So I'd probably say something like "Rule '%s' 'then' '%s' must specficy all role types explicitly or using a variable"